--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts
@@ -14,6 +14,13 @@
 	aliases {
 		mmc2 = &sdio0;
 	};
+
+	/* 25MHz clock for ov5647 */
+	cam_clk: cam_clk {
+		compatible = "fixed-clock";
+		clock-frequency = <25000000>;
+		#clock-cells = <0>;
+	};
 };
 
 &sdio0 {
@@ -47,3 +54,55 @@
 		vddio-supply = <&vcc_1v8>;
 	};
 };
+
+/* Always high for always power on ov5647 */
+&pinctrl {
+	camera_ctrl {
+		cam_gpio1: cam_gpio1 {
+			rockchip,pins = <1 RK_PB5 RK_FUNC_GPIO &pcfg_output_high>;
+		};
+	};
+};
+
+/* I2C port setting for MIPI CSI2 Camera. */
+camera: &i2c4 {
+	status = "okay";
+	clock-frequency = <400000>;
+
+	ov5647: camera@36 {
+		compatible = "ovti,ov5647";
+		reg = <0x36>;
+		clocks = <&cam_clk>;	// This clock needs to set 25MHz.
+		/* pwdn-gpios = <&gpio1 RK_PB5 GPIO_ACTIVE_HIGH>; */
+
+		port {
+			camera_out: endpoint {
+				remote-endpoint = <&csi1_ep1>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
+};
+
+/* Add override for support rkisp1 and MIPI CSI camera */
+&isp0 {
+	status = "okay";
+
+	ports {
+		port@0 {
+			csi1_ep1: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&camera_out>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
+};
+
+&isp0_mmu {
+	status = "okay";
+};
+
+&mipi_dphy_rx0 {
+	status = "okay";
+};
